PCLumiProd/DeriveHFCorrections.py

```diff
@@ -0,0 +1,182 @@
+import sys, os
+from math import exp
+import argparse
+import subprocess
+import ROOT
+
+parser=argparse.ArgumentParser()
+#parser.add_argument("-h", "--help", help="Display this message.")
+parser.add_argument("-f", "--file", default="", help="The path to a cert tree.")
+parser.add_argument("-d", "--dir",  default="", help="The path to a directory of cert trees.")
+parser.add_argument("-r", "--runs", default="", help="Comma separated list of runs.")
+parser.add_argument("--auto", default=False, action="store_true", help="Determine the runs from the certtree")
+parser.add_argument("-l", "--label", default="", help="The label for outputs")
+parser.add_argument('-b', '--batch',   action='store_true', default=False, help="Batch mode (doesn't make GUI TCanvases)")
+
+args=parser.parse_args()
+
+def findRange( hist, cut):
+    gaplist=[]
+    for i in range(1,3600):
+        if hist.GetBinContent(i)<cut:
+            gaplist.append(i)
+    return gaplist
+
+ROOT.gStyle.SetOptStat(0)
+if args.batch is True:
+    ROOT.gROOT.SetBatch(ROOT.kTRUE)
+
+
+corrTemplate=ROOT.TH1F("corrTemplate","",3600,0,3600)
+
+#corrTemplate.SetBinContent(1,a+b*exp(c))
+#for i in range(1,3600):
+#    corrTemplate.SetBinContent(i,b*exp(-(i-2)*c))
+#corrTemplate.GetXaxis().SetRangeUser(0,100)
+
+filename=args.file
+if args.runs!="":
+    runs=args.runs.split(",")
+else:
+    runs=[]
+label=args.label
+
+
+templatefile=ROOT.TFile("HF_template.root")
```

Comment:
Please commit the HF_template.root file as well.  Please put the fill number in the file name and update this script as well.